charmcraft/__init__.py

```diff
@@ -21,9 +21,9 @@
 import pkg_resources
 
 
-def _get_version():
+def _get_version() -> str:
     if os.getenv("SNAP_NAME") == "charmcraft":
-        return os.getenv("SNAP_VERSION")
+        return os.getenv("SNAP_VERSION", "")
```

Comment:
a possibly pedantic point: if `SNAP_NAME` is set then `SNAP_VERSION` should always be set too, right? In that case, how about raising an error instead if it's not set, instead of returning an empty string?

Reply:
Pedantically correct :-)

I'm choosing not to take action now though since the craft-application rebase will change this function anyway.